include/pybind11/cast.h

```diff
@@ -469,6 +469,7 @@ template <typename type> class type_caster<std::reference_wrapper<type>> : publi
     public: \
         static PYBIND11_DESCR name() { return type_descr(py_name); } \
         static handle cast(const type *src, return_value_policy policy, handle parent) { \
+            if (!src) return none().release(); \
```

Comment:
They're essentially the same, but it turns out that `.release()` gives the compiler an opportunity for code size optimization (eliminating the destructor call) in Release mode (as opposed to MinSizeRel, where it doesn't make a difference).

Might need to look into changing the other call sites as well.